src/ResourceManager/Sql/Commands.Sql/Elastic Pools/Cmdlet/NewAzureSqlElasticPool.cs

```diff
@@ -146,25 +173,56 @@ protected override IEnumerable<AzureSqlElasticPoolModel> ApplyUserInputToModel(I
         {
             string location = ModelAdapter.GetServerLocation(ResourceGroupName, ServerName);
             List<AzureSqlElasticPoolModel> newEntity = new List<AzureSqlElasticPoolModel>();
-            newEntity.Add(new AzureSqlElasticPoolModel()
+            AzureSqlElasticPoolModel newModel = new AzureSqlElasticPoolModel()
             {
                 ResourceGroupName = ResourceGroupName,
                 ServerName = ServerName,
+                ElasticPoolName = ElasticPoolName,
                 Tags = TagsConversionHelper.CreateTagDictionary(Tags, validate: true),
                 Location = location,
-                ElasticPoolName = ElasticPoolName,
-                DatabaseDtuMax = MyInvocation.BoundParameters.ContainsKey("DatabaseDtuMax") ? (int?)DatabaseDtuMax : null,
-                DatabaseDtuMin = MyInvocation.BoundParameters.ContainsKey("DatabaseDtuMin") ? (int?)DatabaseDtuMin : null,
-                Dtu = MyInvocation.BoundParameters.ContainsKey("Dtu") ? (int?)Dtu : null,
-                Edition = MyInvocation.BoundParameters.ContainsKey("Edition") ? (DatabaseEdition?)Edition : null,
-                StorageMB = MyInvocation.BoundParameters.ContainsKey("StorageMB") ? (int?)StorageMB : null,
                 ZoneRedundant = MyInvocation.BoundParameters.ContainsKey("ZoneRedundant") ? (bool?)ZoneRedundant.ToBool() : null,
-            });
+                MaxSizeBytes = MyInvocation.BoundParameters.ContainsKey("StorageMB") ? (long?)(StorageMB * Megabytes) : null
+            };
+
+            if (ParameterSetName == DtuPoolParameterSet)
+            {
+                DatabaseEdition? edition = MyInvocation.BoundParameters.ContainsKey("Edition") ? (DatabaseEdition?)Edition : null;
+                if(edition.HasValue)
+                {
+                    newModel.Sku = new Management.Sql.Models.Sku()
+                    {
+                        Name = edition.HasValue ? string.Format("{0}{1}", edition.Value, NormalElasticPoolSkuNamesPostfix) : DefaultPoolSkuName,
+                        Tier = edition.HasValue ? edition.ToString() : null,
```

Comment:
@payiAzure the above two lines don't need a check for `edition.HasValue` since you already checked for that condition in the if-statement that this code resides in

Reply:
@cormacpayne will fix